### PR TITLE
feat(@desktop/wallet2): Added new Tab to display Collectibles

### LINKED ISSUE #3306: update new collectibles UI
collectibles list and view collectibles should look like this https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet?node-id=7%3A6096

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
New features added :
1. Tab which displays the collection
2. Page which display the details of the NFT

Link to StatusQ PR : https://github.com/status-im/StatusQ/pull/407
Link to status-lib PR: https://github.com/status-im/status-lib/pull/30
Link to snim-status-lib PR: https://github.com/status-im/status-lib/pull/11

fixes #3306

### CODE REVIEW

File: src/app/wallet/v2/views/asset_list.nim [hunk 1]
```diff
@@ -1,5 +1,6 @@
-import NimQml, Tables
+import NimQml, Tables, trait_list
 from status/wallet2 import OpenseaAsset
+import NimQml, Tables, trait_list
```
Comment (alaibe) [inline src/app/wallet/v2/views/asset_list.nim:3]: we usualy seperate our import from std import. So maybe add a new import line with trait_list

File: ui/app/AppLayouts/WalletV2/CollectiblesTab.qml [hunk 2]
```diff
@@ -52,6 +54,7 @@ Item {
         }
     }
 
+    // To-do remove
```
Comment (alaibe) [inline ui/app/AppLayouts/WalletV2/CollectiblesTab.qml:57]: nit: is the todo still valid?

File: src/app/wallet/v2/views/asset_list.nim (hunk 1 above)
Comment (Khushboo-dev-cpp) [reply src/app/wallet/v2/views/asset_list.nim]: Done!

File: ui/app/AppLayouts/WalletV2/CollectiblesTab.qml (hunk 2 above)
Comment (Khushboo-dev-cpp) [reply ui/app/AppLayouts/WalletV2/CollectiblesTab.qml]: removed the unused component

File: src/app/wallet/v2/views/collection_list.nim [hunk 3]
```diff
@@ -60,7 +60,12 @@ QtObject:
     CollectionRoles.ImageUrl.int:"imageUrl",
     CollectionRoles.OwnedAssetCount.int:"ownedAssetCount"}.toTable
 
+  proc getCollectionTraitMaxValue(self: CollectionList, index: int, traitType: string): int {.slot.} =
+        let collection = self.collections[index]
+        echo int(collection.trait[traitType].max)
+        return int(collection.trait[traitType].max)
```
Comment (alaibe) [inline src/app/wallet/v2/views/collection_list.nim:66]: nit: indent issue

File: src/app/wallet/v2/views/collection_list.nim [hunk 4]
```diff
@@ -60,7 +60,12 @@ QtObject:
     CollectionRoles.ImageUrl.int:"imageUrl",
     CollectionRoles.OwnedAssetCount.int:"ownedAssetCount"}.toTable
 
+  proc getCollectionTraitMaxValue(self: CollectionList, index: int, traitType: string): int {.slot.} =
+        let collection = self.collections[index]
+        echo int(collection.trait[traitType].max)
```
Comment (alaibe) [inline src/app/wallet/v2/views/collection_list.nim:65]: nit: echo can be removed here

File: src/app/wallet/v2/views/asset_list.nim [hunk 5]
```diff
@@ -22,6 +30,9 @@ QtObject:
   proc newAssetList*(): AssetList =
     new(result, delete)
     result.assets = @[]
+    result.propertiesList = newTraitsList()
+    result.rankingList = newTraitsList()
+    result.statsList = newTraitsList()
```
Comment (alaibe) [inline src/app/wallet/v2/views/asset_list.nim:35]: Not sure here, but should you not also delete the 3 nested lists?

File: ui/app/AppLayouts/WalletV2/components/CollectibleCollection.qml [hunk 6]
```diff
@@ -41,157 +25,75 @@ Item {
...
+                    image.source: model.imageUrl
+                    border.color: Theme.palette.baseColor2
+                    border.width: 1
+                    showLoadingIndicator: true
+                    color: model.backgroundColor
+                    MouseArea {
+                        anchors.fill: parent
+                        onClicked: {
+                            openCollectibleDetailView({collectibleImageUrl:collectionImageUrl,
+                                                          name: model.name,
```
Comment (alaibe) [inline ui/app/AppLayouts/WalletV2/components/CollectibleCollection.qml:78]: nit: indent problem

File: src/app/wallet/v2/views/asset_list.nim [hunk 7]
```diff
@@ -55,16 +66,33 @@ QtObject:
     of AssetRoles.Description: result = newQVariant(asset.description)
     of AssetRoles.Permalink: result = newQVariant(asset.permalink)
     of AssetRoles.ImageUrl: result = newQVariant(asset.imageUrl)
+    of AssetRoles.BackgroundColor: result = newQVariant(if (asset.backgroundColor == ""): "transparent" else: ("#" & asset.backgroundColor))
+    of AssetRoles.Properties:
+        self.propertiesList.traits = @[]
```
Comment (alaibe) [inline src/app/wallet/v2/views/asset_list.nim:71]: Does it mean that each time you access the attributes properties, it also mutate the asset_list view?
If I understand that correctly this is probably wrong that a reader is also mutating internal element of the view right?

If I am correct in my understanding, what about creating an single asset view: which would be set explicitly when viewing a specific asset

wdyt?

**Review by alaibe — CHANGES_REQUESTED** [review]
Nice work, only request changes because of my question on asset_list. If I am wrong, happy to approve

File: src/app/wallet/v2/views/asset_list.nim (hunk 7 above)
Comment (Khushboo-dev-cpp) [reply src/app/wallet/v2/views/asset_list.nim:71]: Hey @alaibe It does not mutate the whole asset_list at all. I have only created the properties, ranking, stats list to be able to pass the data in the OpenseaAsset. If you see the line -
        self.propertiesList.traits = asset.properties
this is only used to update the properties list with the data in the asset. This is done to be able to pass it on to QML as a abstractListItemModel. 

Asset list contains the list of all assets, each asset has lists of properties , ranking and stats and I simply use the 
propertiesList*: TraitsList
     rankingList*: TraitsList
     statsList*: TraitsList
to pass then information already stored in the asset to the UI side.

File: src/app/wallet/v2/views/asset_list.nim (hunk 5 above)
Comment (Khushboo-dev-cpp) [reply src/app/wallet/v2/views/asset_list.nim:35]: definitely . My bad

File: ui/app/AppLayouts/WalletV2/components/CollectibleCollection.qml (hunk 6 above)
Comment (Khushboo-dev-cpp) [reply ui/app/AppLayouts/WalletV2/components/CollectibleCollection.qml:78]: that's just qtCreators indentation. should be okay?

File: src/app/wallet/v2/views/asset_list.nim (hunk 7 above)
Comment (Khushboo-dev-cpp) [reply src/app/wallet/v2/views/asset_list.nim:74]: Addressed all the other review comments :)

**Review by iurimatias — CHANGES_REQUESTED** [review]
@Khushboo-dev-cpp I get this error compiling this branch:

```
/Users/iurimatias/Projects/Status/nim-status-client/src/app/wallet/v2/views/asset_list.nim(17, 1) template/generic instantiation of `QtObject` from here
/Users/iurimatias/Projects/Status/nim-status-client/src/app/wallet/v2/views/asset_list.nim(28, 5) Error: undeclared identifier: 'result'
```

File: src/app/wallet/v2/views/asset_list.nim (hunk 7 above)
Comment (alaibe) [reply src/app/wallet/v2/views/asset_list.nim:71]: Gotcha thanks for clarification. I think the only possible problem of the solution is that it cannot shows traits of 2 assets at the same time

But as this is not part of the design, I guess it's ok!
Sorry for delay

**Pull request merged**